internal/webhooks/test/clusterclass_test.go

```diff
@@ -268,60 +280,63 @@ func TestClusterClassWebhook_Delete(t *testing.T) {
 	ns, err := env.CreateNamespace(ctx, "test-topology-clusterclass-webhook")
 	g.Expect(err).ToNot(HaveOccurred())
 
+	cleanup, err := createTemplates(ns)
+	g.Expect(err).ToNot(HaveOccurred())
+
 	// Create the objects needed for the integration test:
 	// - a two ClusterClasses with all the related templates
 	// - a Cluster using one of the ClusterClasses
-	clusterClass1 := builder.ClusterClass(ns.Name, "class1").
+	clusterClass1 := builder.ClusterClass(ns.Name, clusterClassName1).
 		WithInfrastructureClusterTemplate(
-			builder.InfrastructureClusterTemplate(ns.Name, "inf").Build()).
+			builder.TestInfrastructureClusterTemplate(ns.Name, infrastructureClusterTemplateName1).Build()).
 		WithControlPlaneTemplate(
-			builder.ControlPlaneTemplate(ns.Name, "cp1").
+			builder.TestControlPlaneTemplate(ns.Name, controlPlaneTemplateName1).
 				Build()).
 		WithWorkerMachineDeploymentClasses(
-			*builder.MachineDeploymentClass("md1").
+			*builder.MachineDeploymentClass(workerClassName1).
 				WithInfrastructureTemplate(
-					builder.InfrastructureMachineTemplate(ns.Name, "OLD_INFRA").Build()).
+					builder.TestInfrastructureMachineTemplate(ns.Name, infrastructureMachineTemplateName1).Build()).
 				WithBootstrapTemplate(
-					builder.BootstrapTemplate(ns.Name, "bootstrap1").Build()).
+					builder.TestBootstrapTemplate(ns.Name, bootstrapTemplateName1).Build()).
 				Build(),
-			*builder.MachineDeploymentClass("md2").
+			*builder.MachineDeploymentClass(workerClassName2).
 				WithInfrastructureTemplate(
-					builder.InfrastructureMachineTemplate(ns.Name, "infra1").Build()).
+					builder.TestInfrastructureMachineTemplate(ns.Name, infrastructureMachineTemplateName1).Build()).
 				WithBootstrapTemplate(
-					builder.BootstrapTemplate(ns.Name, "bootstrap1").Build()).
+					builder.TestBootstrapTemplate(ns.Name, bootstrapTemplateName1).Build()).
 				Build()).
 		Build()
 
-	clusterClass2 := builder.ClusterClass(ns.Name, "class2").
+	clusterClass2 := builder.ClusterClass(ns.Name, clusterClassName2).
 		WithInfrastructureClusterTemplate(
-			builder.InfrastructureClusterTemplate(ns.Name, "inf").Build()).
+			builder.TestInfrastructureClusterTemplate(ns.Name, infrastructureClusterTemplateName1).Build()).
 		WithControlPlaneTemplate(
-			builder.ControlPlaneTemplate(ns.Name, "cp1").
+			builder.TestControlPlaneTemplate(ns.Name, controlPlaneTemplateName1).
 				Build()).
 		WithWorkerMachineDeploymentClasses(
-			*builder.MachineDeploymentClass("md1").
+			*builder.MachineDeploymentClass(workerClassName1).
 				WithInfrastructureTemplate(
-					builder.InfrastructureMachineTemplate(ns.Name, "OLD_INFRA").Build()).
+					builder.TestInfrastructureMachineTemplate(ns.Name, infrastructureMachineTemplateName2).Build()).
 				WithBootstrapTemplate(
```

Comment:
Why is workerClassName1 now using infrastructureMachineTemplateName2 while workerClassName2 is using infrastructureMachineTemplateName1?

Reply:
No reason here.